### ISSUE TITLE
bug: apr check (selftest) fails with GH-187 embedding warning

### ISSUE THREAD

**Issue by noahgift** [paiml/aprender#253]
## Summary

\`apr check\` (model self-test) fails for all models with exit code 5, reporting \`Model self-test failed\`. The stderr includes GH-187 embedding warnings suggesting the self-test implementation has a known issue.

## Error Output

\`\`\`
[GH-187] Embedding 'lm_head.weight': shape=[49152, 576], dtype=F32
[GH-187] Embedding 'model.embed_tokens.weight': shape=[49152, 576], dtype=F32
error: Validation failed: Model self-test failed
\`\`\`

## Impact

- 6 selftest checks fail (all 3 models x 2 quant levels)
- The GH-187 prefix suggests this is a known issue

## Reproduction

\`\`\`bash
apr check models/smollm-135m-int8.apr --json
# exit 5
\`\`\`

## Environment

- \`apr 0.2.16 (1fd507b6)\`
- All models affected: SmolLM-135M, Qwen2-0.5B, GPT-2

**Comment by noahgift** [comment]
Likely resolved by the GH-250 fix (realizarcommit 3b1eefb). The selftest runs `model.forward()` which calls `get_tensor_f32()` — this was using GGML Q8_0 dequant on APR Q8 data (completely different binary format), producing garbage logits. With correct APR-native dequant, the forward pass should produce valid logits and pass the softmax/sampler checks. Needs verification with actual model files.
